src/server/src/main/java/io/cassandrareaper/jmx/JmxProxyImpl.java

```diff
@@ -1091,8 +1091,16 @@ private void registerConnectionsGauge() {
   }
 
   @Override
-  public List<Snapshot> listSnapshots() {
+  public List<Snapshot> listSnapshots() throws ReaperException {
```

Comment:
method never throws `ReaperException`

Reply:
`versionCompare()` can throw a ReaperException, which is why it's required to either catch or throw it.

Reply:
my bad.

that's confusing when `UnsupportedOperationException` is the exception noticeably being thrown.

And I don't think `versionCompare(..)` should be throwing a `ReaperException`. what has it to do with Reaper? (it even says it's code taken from stackoverflow.) My expectation was that a method like `versionCompare(..)` would only be throwing a IllegalArgumentException.

And why arn't we using `com.datastax.driver.core.VersionNumber` instead?


maybe these questions are out-of-scope for this PR, as this is code that was previously there?